common/src/main/java/com/wynntils/core/webapi/WebManager.java

```diff
@@ -231,12 +234,69 @@ public static boolean tryLoadItemList() {
                 .useCacheAsBackup()
                 .build());
 
-        handler.dispatch(false);
-
         // Check for success
         return isItemListLoaded();
     }
 
+    public static CompletableFuture<Boolean> tryLoadMap() {
+        if (apiUrls == null || !apiUrls.hasKey("AMainMap")) return CompletableFuture.completedFuture(false);
+
+        final File mapDirectory = new File(API_CACHE_ROOT, "map");
+
+        mapDirectory.mkdirs();
+
+        final CompletableFuture<Boolean> result = new CompletableFuture<>();
+
+        handler.addAndDispatch(new RequestBuilder(apiUrls.get("AMainMap"), "main_map.info")
```

Comment:
```suggestion
        handler.addAndDispatch(new RequestBuilder(apiUrls.get("AMainMap"), "map")
```

This is, afaict, just a marker that we use to distinguish different request. Better make it a bit nicer.